### ISSUE TITLE
Naming convention should support exclusions

### ISSUE THREAD

**Issue by CSchulz** [ajafff/tslint-consistent-codestyle#5]
There are some fixed method names like `toJSON` which is automatically called by `JSON.stringify`, which don't follow the rules of CamelCase.

It would be nice to have the possibilty to define some exclusions for naming convention check.

**Comment by ajafff** [comment]
Agreed, that's a valid use case.

Will try to get this done soon.
In the meantime you could just disable the rule for that line.
Alternatively: add `"prefix": ["toJSON", ""]` to the check to exclude this name.

**Comment by ajafff** [comment]
@CSchulz I added a `filter` property to the configuration of the rule with v1.1.0

for your use case, you can either exclude `toJSON` in the config that specifies the format check
```js
{"type": "default", "filter": "^(?!toJSON$)", "format": "camelCase"} // filter matches everything except "toJSON"
```
or explicitly disable the format check for methods named `toJSON`:
```js
{"type": "method", "filter": "^toJSON$", "format": null}
```

performance-wise I would recommend the latter although the difference should be barely noticeable
